skopt/acquisition.py

```diff
@@ -264,3 +266,86 @@ def gaussian_ei(X, model, y_opt=0.0, xi=0.01, return_grad=False):
         return values, grad
 
     return values
+
+
+def gaussian_entropy_search(X, model, rep_points, n_samples=500,
+                            random_state=None, **kwargs):
+    """
+    Use entropy search to optimize the acquisition function.
+
+    1. It approximates the search space by a set of representer points sampled from
+       a suitable measure such as the "probability of improvement" or
+       "expected improvement".
+
+    2. p_min at a representer point X_i is calculated by the Integral
+       p(f(x)) \prod_{j=1}^N_rep \Theta(f(X_j) - f(X_i)) \\
+       Where P(f(x)) is the GP posterior and \Theta is the heaviside step function.
+       This can be used to calculate the base entropy.
+
+    3. Then for each candidate point, the change in the mean and the covariance
+       of each of these representer points can be computed. This in turn
+       can be used to compute the change in p_min for each of these representer
+       points.
+
+    Parameters
+    ----------
+    * `X` [array-like, shape=(n_samples, n_features)]:
+        Values where the acquisition function should be computed.
+
+    * `model` [sklearn estimator that implements predict with ``return_std``]:
+        The fit estimator that approximates the function through the
+        method ``predict``.
+        It should have a ``return_std`` parameter that returns the standard
+        deviation.
+
+    * `space` [Space instance]:
+        This is required to sample representer points.
+
+    * `n_samples` [int, default 500]:
+        Number of functions to sample from the GP posterior at the
+        representer points to approximate p_min at each of the representer
+        points.
+
+    * `random_state` [int, RandomState instance, or None (default)]:
+        For reproducible results.
+    """
+    rng = np.random.RandomState(random_state)
+    n_rep_points = len(rep_points)
+    n_features = X.shape[1]
+    X_all = np.vstack((rep_points, X))
+    X_all_mean, X_all_cov = model.predict(X_all, return_cov=True)
```

Comment:
The prediction of the mean and covariance of the representer points can be moved outside the function `gaussian_entropy_search` for lbfgs but I prefer to do that later.